File: Python/8py 2/hmldyo.py
"""
|--------------------------------------------------------------------------
| How many lightsabers do you own?
|--------------------------------------------------------------------------
|
"""

def how_many_light_sabers_do_you_own(name=None):
   return 18 if name == 'Zach' else 0


def how_many_light_sabers_do_you_own(name = ''):
   return [0, 18][int(name == "Zach")]


def how_many_light_sabers_do_you_own(name=""):
   return {'Zach': 18}.get(name, 0)


def how_many_light_sabers_do_you_own(name=""):
   return 18 * (name == "Zach")


print(how_many_light_sabers_do_you_own("Zach"), 18)
print(how_many_light_sabers_do_you_own(), 0)
print(how_many_light_sabers_do_you_own("zach"), 0)

File: PHP/8php 1/hmldyo.php
<?php
/*
|--------------------------------------------------------------------------
| How many lightsabers do you own?
|--------------------------------------------------------------------------
|
*/

function howManyLightsabersDoYouOwn($name='') {
   return $name == 'Zach' ? 18 : 0;
}


function howManyLightsabersDoYouOwn($str = "") {
   return preg_match_all("/Zach/",$str) * 18;
}


function howManyLightsabersDoYouOwn() {
   $argc = func_get_args();
   return isset($argc[0]) && $argc[0] === 'Zach' ? 18 : 0;
}


print_r(howManyLightsabersDoYouOwn() . ' ' . 0 . '
');
print_r(howManyLightsabersDoYouOwn("Adam") . ' ' . 0 . '
');
print_r(howManyLightsabersDoYouOwn("Zach") . ' ' . 18 . '
');